libminifi/include/core/PropertyValue.h

```diff
@@ -138,26 +150,28 @@ class PropertyValue : public state::response::ValueNode {
    */
   template<typename T>
   auto operator=(const T ref) -> typename std::enable_if<std::is_same<T, std::string>::value,PropertyValue&>::type {
```

Comment:
I believe `T` if not explicitly provided will always be deduced to a non-reference type, good question what the author wanted to prevent (or facilitate) with this solution

Reply:
Yes, that is correct, `T&&` should bind to `T`, and is_same handles `volatile`/`const`. Still, the simple overload should be simpler?

Reply:
I would not touch this now, as it is part of the API, after 0.8.0 we should revisit this, as well as the whole codebase in general